### PR TITLE
[Experience] Change Exp Calculations to be 64 bit where needed.

### PR DESCRIPTION
**Pull request by Aeadoin**
Before, would see potential overflows causing Experience Loss messages

image

After, no longer sees these messages, and can crank xp mods up insanely high without it breaking.

image

### CODE REVIEW

File: zone/exp.cpp [hunk 1]
```diff
@@ -202,7 +202,7 @@ uint32 Client::CalcEXP(uint8 conlevel) {
 	return in_add_exp;
 }
 
-uint32 Client::GetExperienceForKill(Mob *against)
+uint64 Client::GetExperienceForKill(Mob *against)
 {
 #ifdef LUA_EQEMU
 	uint32 lua_ret = 0;
```
Comment (Akkadius) [inline zone/exp.cpp:208]: `lua_ret` will need to be adjusted as well for the associated method

```cpp
uint32 LuaParser::GetExperienceForKill(Client *self, Mob *against, bool &ignoreDefault)
{
	uint32 retval = 0;
	for (auto &mod : mods_) {
		mod.GetExperienceForKill(self, against, retval, ignoreDefault);
	}
	return retval;
}
```

This is for the lua mod system

**Review by Akkadius — APPROVED** [review]
LGTM. Few minor comments

File: zone/exp.cpp (hunk 1 above)
Comment (Aeadoin) [reply zone/exp.cpp]: Taken care of.

**Review by Kinglykrab — APPROVED** [review]

File: zone/lua_client.cpp [hunk 2]
```diff
@@ -3418,9 +3418,10 @@ luabind::scope lua_register_client() {
 	.def("SetDeity", (void(Lua_Client::*)(int))&Lua_Client::SetDeity)
 	.def("SetDuelTarget", (void(Lua_Client::*)(int))&Lua_Client::SetDuelTarget)
 	.def("SetDueling", (void(Lua_Client::*)(bool))&Lua_Client::SetDueling)
-	.def("SetEXP", (void(Lua_Client::*)(uint32,uint32))&Lua_Client::SetEXP)
-	.def("SetEXP", (void(Lua_Client::*)(uint32,uint32,bool))&Lua_Client::SetEXP)
+	.def("SetEXP", (void(Lua_Client::*)(uint64,uint64))&Lua_Client::SetEXP)
+	.def("SetEXP", (void(Lua_Client::*)(uint64,uint64,bool))&Lua_Client::SetEXP)
 	.def("SetEXPEnabled", (void(Lua_Client::*)(bool))&Lua_Client::SetEXPEnabled)
+
```
Comment (Kinglykrab) [inline zone/lua_client.cpp:3424]: Extra new line here.

**Pull request merged**